Integrations/integration-LogRhythm.yml

```
@@ -718,6 +843,171 @@ script:
+
+        if (mdRaw['-i']) {
+            delete mdRaw['-i'];
+        }
+
+        fixDictionaryKeys(mdRaw);
+
+        mdRaw.forEach(function(row){
+            hostsFieldsToRemove.forEach(function(field){
+                delete row[field];
+            });
+        });
+
+        var md = tableToMarkdown('LogRhythm hosts for entity ID: ' + args['entity-id'],  mdRaw);
+
+        var ec = {};
+        ec["LogRhythm.Host(val.ID && val.ID == " + mdRaw['ID'] + ")"] = mdRaw;
+
+        return {
+            Type: entryTypes.note,
+            Contents: jsResponse,
+            ContentsFormat: formats.json,
+            HumanReadable: md,
+            EntryContext: ec
+        };
+
+    }
+
+    function addHost(dataArgs){
+        var dataObj = {
+            Username: params.Credentials.identifier,
+            Password: params.Credentials.password,
+            Hostname: dataArgs['host-name'],
+            EntityId: dataArgs['entity-id'],
+            ShortDesc: dataArgs['short-description'],
+            LongDesc: dataArgs['long-description'],
+            DateUpdated: new Date().toISOString(),
+            RiskThreshold: dataArgs['risk-threshold'],
+            OsType: dataArgs['os-type'],
+            OsVersion: dataArgs['os-version'],
+            ThreatThreshold: dataArgs['threat-threshold']
+        };
+
+        var soapActionURIUaddHost = 'http://www.logrhythm.com/webservices/HostService/AddHost';
+
+        var envelope = fixTemplate(tmplAddHost1, dataObj);
+
+        if(dataArgs['ip-address']){
+            envelope += fixTemplate(tmplAddHostIdentifier, {Type: 'IPAddress', Value: dataArgs['ip-address']});
+        }
+        if(dataArgs['dns-name']){
+            envelope += fixTemplate(tmplAddHostIdentifier, {Type: 'DNSName', Value: dataArgs['dns-name']});
+        }
+        if(dataArgs['windows-name']){
+            envelope += fixTemplate(tmplAddHostIdentifier, {Type: 'WindowsName', Value: dataArgs['windows-name']});
+        }
+
+        envelope += fixTemplate(tmplAddHost2, dataObj);
+
+        var jsResponse = parseResponse(sendRequest(envelope, soapActionURIUaddHost, logRhythmHostAPI));
+
+        var mdRaw = dq(jsResponse, 'Envelope.Body.AddHostResponse.AddHostResult')
+
+        if (mdRaw['Succeeded'] == 'true'){
+            var host = {};
+            host["ShortDescription"] = dataObj['ShortDesc'];
+            host["LongDescription"] = dataObj['LongDesc'];
+            host["HostID"] = mdRaw['DataID'];
+            host["RecordStatus"] = 'NewRecord';
+            host["DateUpdated"] = dataObj['DateUpdated'];
+            host["ThreatThreshold"] = dataObj['ThreatThreshold'];
+            host["HostName"] = dataObj['Hostname'];
+            host["OSVersion"] = dataObj['OsVersion'];
+            host["RiskThreshold"] = dataObj['RiskThreshold'];
+            host["EntityID"] = dataObj['EntityId'];
+            host["OSType"] = dataObj['OsType'];
+
+            var ec = {};
+            ec["LogRhythm.Host"] = host;
```

use DT to avoid duplications:
```suggestion
            ec["LogRhythm.Host(val.HostID && val.HostID == obj.HostID)"] = host;
```

---

why should i check for duplicates when adding new host?